### PR TITLE
Add Golang support.

### PR DESCRIPTION
**Pull request by patricklundquist**
Adds support for Golang

### CODE REVIEW

**Comment by tarzanek** [comment]
Hello Patrick, lovely, lgtm and very nice work, just one small thing before we can accept, I need you to either say here you're OK with OCA for this patch,
OR check  http://www.oracle.com/technetwork/community/oca-486395.html  and print, sign, scan and email back a signed copy to get registered with Oracle (they own trademark for OpenGrok)
thank you
Lubos

P.S. actually, there is one thing which could be added which is junit tests - add a sample go file into sources so tests will pick them up and recognize defs or full hits for go analyzer - but we can put it into next pull request (and I owe such request to our junits for scala)

File: src/org/opensolaris/opengrok/analysis/golang/Consts.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*Patrick Lundquist
```
Comment (tarzanek) [inline src/org/opensolaris/opengrok/analysis/golang/Consts.java]: ok, so actually just one small prob, header has to be proper, please move the name to @author section, this line will break checkstyle ... ;)

**Comment by patricklundquist** [comment]
Hi Lubos, I'm OK with OCA for this patch. 
Thanks!
Patrick

**Comment by tarzanek** [comment]
thank you sir, merging!

**Pull request merged**